### PR TITLE
DOC: Updates to installation documentation

### PR DESCRIPTION
**Pull request by ljlamarche**
# Description

This PR makes some updates and clarifications to the apexpy installation instructions, specifically troubleshooting unusual cases.  Addresses https://github.com/aburrell/apexpy/issues/101 by specifying apexpy has not been tested with versions of gfortran below v7.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- This change is a documentation update

# How Has This Been Tested?

Compiled sphinx documentation locally.

**Test Configuration**:
* Operating system: MacOS
* Python version number: 3.9

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``Changelog.rst``, summarising the changes
- [x] Add yourself to ``AUTHORS.rst`` and ``.zenodo.json``

### CODE REVIEW

**Comment by ljlamarche** [comment]
Is numpy still required to be pre-installed with the new build system?

**Comment by aburrell** [comment]
I think not.  I believe it will grab it, and just check if you have it locally.

From: Leslie Lamarche ***@***.***>
Reply-To: aburrell/apexpy ***@***.***>
Date: Thursday, December 1, 2022 at 12:42
To: aburrell/apexpy ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [aburrell/apexpy] DOC: Updates to installation documentation (PR #102)


Is numpy still required to be pre-installed with the new build system?

File: docs/installation.rst [hunk 1]
```diff
@@ -83,7 +92,7 @@ seemly successful installation with apexpy reporting that fortranapex cannot be
 imported.
 
 ::
-   LDFLAGS="-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib ${LDFLAGS}" /opt/local/bin/python3.10 setup.py install
+   LDFLAGS="-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib ${LDFLAGS}" python setup.py install
```
Comment (aburrell) [inline docs/installation.rst:95]: Should we stop recommending using setup.py in the installation docs?
```suggestion
   LDFLAGS="-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib ${LDFLAGS}" pip .
```

Comment (ljlamarche) [reply docs/installation.rst:95]: Probably, I wasn't sure about that one... A carry-over copy-paste?

Comment (aburrell) [reply docs/installation.rst:95]: Given the other flags I've used with pip, I think it will work.

Comment (aburrell) [reply docs/installation.rst:95]: Or maybe it would go with the build command (more likely).

**Comment by aburrell** [comment]
The test failure issues are fixed in #98

File: docs/installation.rst (hunk 1 above)
Comment (ljlamarche) [reply docs/installation.rst:95]: I'm not sure, I don't think I ever encountered this issue...

Comment (aburrell) [reply docs/installation.rst:95]: Let's leave it alone then, since I am not certain 😨

Comment (ljlamarche) [reply docs/installation.rst:95]: Ugg, I don't particularly want to leave in a recommendation for people to use setup.py - everything I've read seems to indicate that's a bad idea now.  Can we use your original suggestion but move it down to the GitHub section?  Probably should move it regardless.  Alternatively we could create an archive of "installation commands people have reported working but we're not entirely sure why"?

**Comment by ljlamarche** [comment]
Does apexpy require just gfortran or also gcc?  Or do you only need gcc if you're building it locally?

File: docs/installation.rst (hunk 1 above)
Comment (aburrell) [reply docs/installation.rst:95]: I deeply feel that: "installation commands people have reported working but we're not entirely sure why" is appropriate.

**Comment by aburrell** [comment]
> Does apexpy require just gfortran or also gcc? Or do you only need gcc if you're building it locally?

You need gcc and gfortran if you're installing using anything other than a wheel.

**Comment by ljlamarche** [comment]
@aburrell , can you read through this version of the installation page?  It's supposed to be geared towards what the installation process WILL be in v2.0.0, but I wasn't sure about some of it.

File: docs/installation.rst [hunk 2]
```diff
@@ -30,89 +29,90 @@ The package has been tested with the following setups (others might work, too):
...
+the following steps.  `Gfortran <https://gcc.gnu.org/wiki/GFortran>`_ is a free
+compiler that can be installed, if one is not already available on your system.
+If you are installing this or MinGW in Windows, make sure you install it
+**after** installing the Windows Microsoft C++ Build tools.  The apexpy
+installation has been tested successfully with gfortran 7 and some more recent
+versions.  Earlier versions of gfortran may not work and are not recommended.
 
-Installation Issues
-^^^^^^^^^^^^^^^^^^^
+Installation also requires gcc. `GCC <https://gcc.gnu.org/>`_ is a free compiler
```
Comment (aburrell) [inline docs/installation.rst:49]:
```suggestion
Installation also requires a C compiler of the same type as the fortran compiler. `GCC <https://gcc.gnu.org/>`_ is a free compiler that works goth Gfortran and
```

File: docs/installation.rst [hunk 3]
```diff
@@ -30,89 +29,90 @@ The package has been tested with the following setups (others might work, too):
...
+that can be installed from a variety of sources and standard package managers.
```
Comment (aburrell) [inline docs/installation.rst:50]:
```suggestion
can be installed from a variety of sources and standard package managers.
```

File: docs/installation.rst [hunk 4]
```diff
@@ -30,89 +29,90 @@ The package has been tested with the following setups (others might work, too):
...
-locally from the cloned Git repository.  Issues on Mac OS have also been
-encountered when using clang for ``CC`` alongside gfortran.  This resulted in a
-seemly successful installation with apexpy reporting that fortranapex cannot be
-imported.
+Install without Wheels
+^^^^^^^^^^^^^^^^^^^^^^
 
-::
-   LDFLAGS="-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib ${LDFLAGS}" /opt/local/bin/python3.10 setup.py install
+Many times, skipping building wheels locally will solve installation problem,
```
Comment (aburrell) [inline docs/installation.rst:91]:
```suggestion
Many times, skipping building wheels locally will solve installation problems,
```

**Review by aburrell — COMMENTED** [review]
This looks good!  I changed a few things, but thought the section flowed well and contains a lot of the positive results in the installation problem issues.

**Comment by ljlamarche** [comment]
Awesome, I accepted your suggestions!  Feel free to merge when convenient!

**Review by aburrell — APPROVED** [review]

**Pull request merged**